File: 04-solver-platform/api/app/schemas/color_policy.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime


class ColorPolicyOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    color_policy_id: str
    season_id:       str
    slot_colors:     dict[str, Optional[str]]
    palette:         list[str]
    tod_formats:     dict[str, Optional[str]]
    dh_format:       Optional[str] = None
    updated_at:      Optional[datetime] = None


class ColorPolicyUpdate(BaseModel):
    """All fields optional — only supplied fields are updated."""
    slot_colors: Optional[dict[str, Optional[str]]] = None
    palette:     Optional[list[str]]                = None
    tod_formats: Optional[dict[str, Optional[str]]] = None
    dh_format:   Optional[str]                      = None

File: 04-solver-platform/frontend/src/entities/ColorPolicy.ts
export default interface ColorPolicy {
  color_policy_id: string
  season_id:       string
  slot_colors:     Record<string, string | null>
  palette:         string[]
  tod_formats:     Record<string, string | null>
  dh_format:       string | null
  updated_at:      string | null
}
